python/tvm/relay/op/contrib/arm_compute_lib.py

```diff
@@ -248,8 +242,11 @@ def conv2d(attrs, args):
     return True
 
 
-def qnn_conv2d(attrs, args):
+# TODO check why no annotation is here
```

Comment:
Could you elaborate on this TODO?

Reply:
Removed TODO comments

Reply:
Just an fyi, the reason qnn.conv2d wasn't registered is because as an individual operator it isn't supported. It needs to be accompanied with a requantize op.

Reply:
What was the reason to have a pattern and an operator annotations for nn.dense and nn.conv2d then? Both patterns happily accepts single ops.

Reply:
nn.dense and nn.conv2d will work as a single op, we have all the data we need to create the accompanying ACL layer. qnn.dense and qnn.conv2d are subtly different in that we also require a qnn.requantize op, this op allows us to extract the output quantization info in order to create the layer in ACL. Hope that makes sense?

Reply:
Still not clear. Currently patterns allow `nn.dense` and `nn.conv2d` as single ops without any additional operator annotation. However, there is a difference is in arguments: annotations accept all variants of arguments but patterns restrict second argument to constant. So if operator is a single operator we are more lenient to its arguments. Could you elaborate this discordance please?

Reply:
That's a good point, the single op annotation will not check whether certain inputs are constant. We should probably fix this in another pr.

Reply:
@lhutton1 Agree. All constant args should be changed to var in `nn.conv2d`, `qnn.conv2d`, `nn.dense`, `qnn.dense` and operator annotations removed for `nn.conv2d` and `nn.dense`. Correct?

Reply:
The patterns use `is_constant` for a reason - if I recall correctly it allows us to access the data at compile time, I believe they are treated differently to vars. My thought process was to keep the patterns and annotations as they are and simply add checks in `nn.conv2d` and `nn.dense` to make sure that certain args i.e. kernel are indeed constant, if not we return false and the operator isn't annotated?

Reply:
@lhutton1 Could you point me where exactly on compile time we depend on constant type of the second argument of `nn.conv2d`, `qnn.conv2d`, `nn.dense`, or `qnn.dense`, please?

Reply:
Its not too obvious because the JSON codegen module and metadata module take care of constant serialization, I can't remember exactly how this works. Here you can see a constant being added to the JSON representation: https://github.com/apache/incubator-tvm/blob/main/src/relay/backend/contrib/codegen_json/codegen_json.h#L250 . One example in ACL codegen is we convert the layout of constants to the correct format for ACL: https://github.com/apache/incubator-tvm/blob/main/src/relay/backend/contrib/arm_compute_lib/codegen.cc#L340, hope that helps?

Reply:
@lhutton1 Still not clear. Is the second arg needs to be of constant type just to eliminate the necessity to convert it on run time when JSON serialization (with actual data available) happens? But in case of nn.conv2d typical kernel size is small and sort of negligible. Am I correct here? Or some other considerations were used?

Reply:
Apologies if I'm misunderstanding, so you want to allow the possibility of inputting weights at runtime? If that's the case, why wouldn't they be available at compile time? I think the only consideration was that since weights are normally constant we can optimize conversion to the correct layout ahead of time, avoiding the need to do this at runtime.

Reply:
@lhutton1 Thank you very much for clarifications. I am resolving this.